buildbot_effects/__init__.py

```diff
@@ -180,6 +180,10 @@ def run_effects(
     env["IN_HERCULES_CI_EFFECT"] = "true"
     env["HERCULES_CI_SECRETS_JSON"] = "/run/secrets.json"
     env["NIX_BUILD_TOP"] = "/build"
+    env["TEMPDIR"] = "/tmp"
+    env["TMPDIR"] = "/tmp"
+    env["TEMP"] = "/tmp"
+    env["TMP"] = "/tmp"
```

Comment:
I would like to keep only TMPDIR here. I consider the other ones more of a historical artifact.

Reply:
Good point. I reset every one that matched `/tmp/nix-shell...` - didn't know which one it was that made this tool behave badly. Made me think, should we re-set the environment and only carry relevant stuff over into the effect, like PATH and the things we set in the runner? There's a _lot_ of crud on the environment from `nix develop`, like stale CFLAGS and so on.

Reply:
Do we not have a minimal env here? I don't recall the details. In fact we only should set our own environment variables and inherit nothing implicit.